contracts/token/RarePizzas.sol

```diff
@@ -0,0 +1,265 @@
+        __Ownable_init();
+        __ReentrancyGuard_init();
+        __ERC721_init('Rare Pizzas', 'PIZZA');
+
+        if (rarePizzasBoxContract != address(0)) {
+            _rarePizzasBoxContract = IRarePizzasBox(rarePizzasBoxContract);
+        }
+
+        _contractURI = 'https://raw.githubusercontent.com/PizzaDAO/pizza-smartcontract/master/data/opensea_pizza_metadata.mainnet.json';
+    }
+
+    // IOpenSeaCompatible
+    function contractURI() public view virtual override returns (string memory) {
+        // Metadata provided via github link so that it can be updated or modified
+        return _contractURI;
+    }
+
+    // IRarePizzas
+
+    function isRedeemed(uint256 boxTokenId) public view override returns (bool) {
+        return _redeemedBoxTokenAddress[boxTokenId] != address(0);
+    }
+
+    function redeemRarePizzasBox(uint256 boxTokenId) public override nonReentrant {
+        require(_msgSender() == _rarePizzasBoxContract.ownerOf(boxTokenId), 'caller must own box');
+        _redeemRarePizzasBox(_msgSender(), boxTokenId);
+    }
+
+    // IRarePizzasBox overrides
+
+    function getBitcoinPriceInWei() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getBitcoinPriceInWei();
+    }
+
+    function getPrice() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getPriceInWei();
+    }
+
+    function getPriceInWei() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getPriceInWei();
+    }
+
+    function maxSupply() public view virtual override returns (uint256) {
+        return MAX_TOKEN_SUPPLY;
+    }
+
+    function purchase() public payable virtual override nonReentrant {
+        // redirect to the box contract
+        _rarePizzasBoxContract.purchase();
+    }
+
+    // IOrderAPICallback
+
+    function fulfillResponse(bytes32 request, bytes32 result) public virtual override {
+        // TODO: verify should expect the client and not the EOA of the node
+        require(_msgSender() == address(_orderAPIClient), 'caller not order api');
+        require(_renderRequests[request] != address(0), 'valid request must exist');
+
+        address requestor = _renderRequests[request];
+        uint256 boxTokenId = _renderTokenIds[request];
+
+        _internalMintPizza(requestor, boxTokenId, result);
+    }
+
+    // IERC721 Overrides
+
+    /**
+     * @dev See {IERC721Metadata-tokenURI}.
+     */
+    function tokenURI(uint256 tokenId)
+        public
+        view
+        virtual
+        override(ERC721Upgradeable, IERC721MetadataUpgradeable)
+        returns (string memory)
+    {
+        require(_exists(tokenId), 'does not exist, paisano');
+        return string(abi.encodePacked(_uriBase, _base58Encode(_tokenPizzaArtworkURIs[tokenId])));
+    }
+
+    // IRarePizzasAdmin
+
+    function redeemRarePizzasBoxForOwner(uint256 boxTokenId) public virtual override onlyOwner {
```

Comment:
we might want to come up with something better here.  it would suck to have to call this a thousand times